src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -288,27 +284,27 @@ void getRollupCountForCriteriaGroupsGeneratesAndRunsAQuery() {
             .projectResource(new GoogleProjectResource().googleProjectId("project123"))
             .name("dataset123")
             .id(UUID.randomUUID());
-    TablePointer tablePointer = TablePointer.fromRawSql("table-name");
-    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
-    Query query =
-        new Query(
-            List.of(
-                new FieldVariable(new FieldPointer(tablePointer, "column-name"), tableVariable)),
-            List.of(tableVariable));
+    var settings = new SnapshotBuilderSettings();
+    when(snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(dataset.getId()))
+        .thenReturn(settings);
+    Query query = mock(Query.class);
     var criteriaQueryBuilderMock = mock(CriteriaQueryBuilder.class);
     when(datasetService.retrieve(dataset.getId())).thenReturn(dataset);
-    when(queryBuilderFactory.criteriaQueryBuilder(any(), any()))
+    when(queryBuilderFactory.criteriaQueryBuilder("person", settings))
         .thenReturn(criteriaQueryBuilderMock);
-    when(criteriaQueryBuilderMock.generateRollupCountsQueryForCriteriaGroupsList(any()))
+    var criteriaGroups = List.of(List.of(new SnapshotBuilderCriteriaGroup()));
+    when(criteriaQueryBuilderMock.generateRollupCountsQueryForCriteriaGroupsList(criteriaGroups))
         .thenReturn(query);
-    when(azureSynapsePdao.runQuery(
-            eq(query.renderSQL(QueryTestUtils.createContext(CloudPlatform.AZURE))), any()))
```

Comment:
Writing the test in this way makes the test dependent on the behavior of `renderSQL()`. But since that method isn't being tested here, I think it's better to mock this behavior to narrow the test focus.